stdlib/lang-parser/parserModule/parser.bal

```diff
@@ -0,0 +1,1282 @@
+        };
+        return pkNode;
+    }
+
+    # Method to consume the token from the token buffer if the token matches the expected token.
+    # If the token doesnt match , panic recovery method will be invoked.
+    # 
+    # +mToken - token type.
+    # +rule - grammar rule to which the token should be parsed.
+    # +return - Token.
+    function matchToken(int mToken, NodeKind rule) returns Token {
+        if (self.LAToken(1) == mToken) {
+            Token currToken = self.parserBuffer.consumeToken();
+            return currToken;
+        } else {
+            if (self.LAToken(1) == LEXER_ERROR_TOKEN) {
+                log:printError(self.LookaheadToken(1).lineNumber + ":" + self.LookaheadToken(1).startPos + ": expected " +
+                tokenNames[mToken] + ";found an invalid token '" + self.LookaheadToken(1).text + "'");
+            } else {
+                log:printError(self.LookaheadToken(1).lineNumber + ":" + self.LookaheadToken(1).startPos + ": expected " +
+                tokenNames[mToken] + "; found '" + self.LookaheadToken(1).text + "'");
+            }
+
+            Token panicToken = self.panicRecovery(mToken, rule);
+            return panicToken;
+        }
+    }
+
+    # Error recovery : token  insertion.
+    # The expected token will be inserted inplace of the mismatched token.
+    # 
+    # +mToken - token type of the expected token.
+    # +return - Token.
+    function insertToken(int mToken) returns Token {
+        log:printError(tokenNames[mToken] + " inserted");
+        return {
+            tokenType: mToken,
+            text: tokenNames[mToken],
+            startPos: -1,
+            endPos: -1,
+            lineNumber: 0,
+            index: -1,
+            whiteSpace: ""
+        };
+    }
+
+    # Error recovery : delete token.
+    # Mismatched token is consumed and added to the error token list.
+    # 
+    # +return - Token
+    function deleteToken() returns Token {
+        Token invalidToken = self.parserBuffer.consumeToken();
+        log:printError(invalidToken.lineNumber + ":" + invalidToken.startPos + ": invalid token '" +
+        invalidToken.text + "'");
+        self.errTokens[self.errCount] = invalidToken;
+        self.errCount += 1;
+        self.invalidOccurence = true;
+        return invalidToken;
+    }
+
+    # Panic error recovery
+    # Each grammar rule will have a specific error recovery.
+    # Here all the token will be removed and added to the error token list until we meet a terminal token
+    # 
+    # +mToken - token type of the expected token.
+    # +rule - grammar rule which the token should be parsed.
+    # +return - Token
+    function panicRecovery(int mToken, NodeKind rule) returns Token {
+        boolean panicMode = true;
+        self.errorRecovered = false;
+
+        if (rule == VAR_DEF_STATEMENT_NODE) {
+            if (mToken == SEMICOLON) {
+                Token insertSemi = self.insertToken(mToken);
+
```

Comment:
Do we need to have a newline here? Seems like the logic is clear without the newline. Same in line 153, 182, etc.